addons/escoria-core/game/core-scripts/esc_item.gd

```diff
@@ -516,31 +520,53 @@ func get_animation_player() -> Node:
 #
 # **Returns** The interaction position
 func get_interact_position() -> Vector2:
-	var multiple_positions_found = false
+	var pos_2d_count:int = 0
+	var pos_2d_position = null
+	
+	var esclocation_count = 0
+	var esclocation_position = null
+
+	var interact_count = 0
 	var interact_position = null
+	
 	for c in get_children():
 		if c is Position2D:
-			if interact_position != null:
-				multiple_positions_found = true
-			interact_position = c.global_position
-			# Use node if explicitly called interact_position
-			if c.get_name() == "interact_position":
-				multiple_positions_found = false
-				break
-
-	if interact_position == null and collision != null:
-		interact_position = collision.global_position
+			# Identify any Postion2D nodes
+			if c.is_class("ESCLocation"):
+				esclocation_count += 1
+				esclocation_position = c.global_position
+			elif c.is_class("ESCInteractionLocation"):
+				interact_count += 1
+				interact_position = c.global_position
+			else:
+				# This will catch all other Position2D related nodes
+				# including dialog locations and native Position2D nodes.
+				pos_2d_count += 1
+				pos_2d_position = c.global_position				
+	
+	if interact_position == null and \
+	esclocation_position == null and collision != null:
```

Comment:
```suggestion
		esclocation_position == null and collision != null:
```

Reply:
Only the "collision" works with "is_instance_valid", the other two are coordinates. Updated to "! is_instanace_valid(collision)"